classes/form/reportfilter_form.php

```diff
@@ -33,27 +33,44 @@ public function definition() {
         global $DB;
 
         $mform = $this->_form;
-        $errortypes = $this->_customdata['errortypes'];
+        $reportid = $this->_customdata['reportid'];
         $eventnames = $this->_customdata['eventnames'];
-        $responses = $this->_customdata['responses'];
         $resend = $this->_customdata['resend'];
 
+        if ($reportid == XAPI_REPORT_ID_ERROR) {
+            $errortypes = $this->_customdata['errortypes'];
+            $responses = $this->_customdata['responses'];
+        } elseif ($reportid == XAPI_REPORT_ID_HISTORIC) {
+            $eventcontexts = $this->_customdata['eventcontexts'];
+        }
+
         $mform->addElement('hidden', 'resend');
         $mform->setType('resend', PARAM_BOOL);
         $mform->setDefault('resend', $resend);
 
-        $mform->addElement('select', 'errortype', get_string('errortype', 'logstore_xapi'), $errortypes);
-        $mform->addElement('select', 'eventname', get_string('eventname', 'logstore_xapi'), $eventnames);
-        $mform->addElement('select', 'response', get_string('response', 'logstore_xapi'), $responses);
+        if ($reportid == XAPI_REPORT_ID_ERROR) {
+            $mform->addElement('select', 'errortype', get_string('errortype', 'logstore_xapi'), $errortypes);
+        }
+
+        $eventnameselect = $mform->addElement('select', 'eventnames', get_string('eventname', 'logstore_xapi'), $eventnames);
+        $eventnameselect->setMultiple(true);
+
+        if ($reportid == XAPI_REPORT_ID_ERROR) {
+            $mform->addElement('select', 'response', get_string('response', 'logstore_xapi'), $responses);
+        } elseif ($reportid == XAPI_REPORT_ID_HISTORIC) {
+            $mform->addElement('text', 'fullname', get_string('user', 'logstore_xapi'));
```

Comment:
entering a username does not filter the results
![image](https://user-images.githubusercontent.com/51955245/81838619-449ef300-953e-11ea-9592-e5062476ee14.png)

Reply:
I have fixed this if you wish to test

Reply:
It's working now 👍